masonry/src/contexts.rs

```diff
@@ -764,59 +767,48 @@ impl_context_method!(
         ///
         /// The hovered status is computed from the widget's layout rect. In a
         /// container hierarchy, all widgets with layout rects containing the
-        /// mouse position have hovered status.
+        /// pointer position have hovered status.
+        ///
+        /// If the pointer is [captured], then only that widget and its parents
+        /// can have hovered status. If the pointer is captured but not hovering
+        /// over the captured widget, then no widget has the hovered status.
         ///
-        /// Discussion: there is currently some confusion about whether a
-        /// widget can be considered hovered when some other widget has captured the
-        /// pointer (for example, when clicking one widget and dragging to the
-        /// next). The documentation should clearly state the resolution.
+        /// [captured]: crate::doc::doc_06_masonry_concepts#pointer-capture
         pub fn is_hovered(&self) -> bool {
             self.widget_state.is_hovered
         }
 
-        /// Whether the pointer is captured by this widget.
+        /// Whether the mouse is [captured] by this widget.
         ///
-        /// See [`capture_pointer`] for more information about pointer capture.
-        ///
-        /// [`capture_pointer`]: EventCtx::capture_pointer
+        /// [captured]: crate::doc::doc_06_masonry_concepts#pointer-capture
         pub fn has_pointer_capture(&self) -> bool {
             self.global_state.pointer_capture_target == Some(self.widget_state.id)
         }
 
-        /// The focus status of a widget.
+        /// The [text focus] status of a widget.
+        ///
+        /// The focused widget is the one that receives keyboard events.
         ///
         /// Returns `true` if this specific widget is focused.
         /// To check if any descendants are focused use [`has_focus`].
         ///
-        /// Focus means that the widget receives keyboard events.
-        ///
-        /// A widget can request focus using the [`request_focus`] method.
-        /// It's also possible to register for automatic focus via [`register_for_focus`].
-        ///
-        /// If a widget gains or loses focus it will get a [`Update::FocusChanged`] event.
-        ///
-        /// Only one widget at a time is focused. However due to the way events are routed,
-        /// all ancestors of that widget will also receive keyboard events.
-        ///
-        /// [`request_focus`]: EventCtx::request_focus
-        /// [`register_for_focus`]: UpdateCtx::register_for_focus
-        /// [`Update::FocusChanged`]: crate::Update::FocusChanged
+        /// [text focus]: crate::doc::doc_06_masonry_concepts#text-focus
         /// [`has_focus`]: Self::has_focus
         pub fn is_focused(&self) -> bool {
             self.global_state.focused_widget == Some(self.widget_id())
         }
 
-        /// The (tree) focus status of a widget.
+        /// Whether this widget or any of its descendants is focused.
```

Comment:
The verb should agree with the plurality of the nearest verb, which is descendants, which is plural. So this should be "are focused"

Reply:
Nope, "any of its descendants" is singular.

Reply:
Nope, "any of" is plural. "one of" would be the singular version.

Reply:
Huh... I just checked, and apparently it is for enumerable nouns. TIL.